handlers/transcode.go

```diff
@@ -72,9 +68,179 @@ func (d *CatalystAPIHandlersCollection) TranscodeSegment() httprouter.Handle {
+		}
+		if err != nil {
+			return err
+		}
+		_, err = mw.Write(line)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// RunTranscodeProcess starts `MistLivepeeerProc` as a subprocess to transcode inputStream into renditionsStream.
+func RunTranscodeProcess(mistClient clients.MistAPIClient, request TranscodeSegmentRequest) error {
+	inputStream, renditionsStream := config.GenerateStreamNames()
+	if err := mistClient.AddStream(inputStream, request.SourceFile); err != nil {
+		return fmt.Errorf("error adding stream to Mist: %s", err)
+	}
+
+	configPayload, err := json.Marshal(configForSubprocess(request, inputStream, renditionsStream))
+	if err != nil {
+		return fmt.Errorf("ProcLivepeerConfig json encode: %s", err)
+	}
+	args := string(configPayload)
+
+	transcodeCommand := exec.Command(config.PathMistProcLivepeer, args, "--debug", "8", "--kickoff")
+
+	var stdout, stderr bytes.Buffer
+	stderrPipe, err := transcodeCommand.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("Failed to open stderr pipe: %s", err)
+	}
+	stdoutPipe, err := transcodeCommand.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("Failed to open stdout pipe: %s", err)
+	}
+
+	// Start the Transcode Command asynchronously - we call Wait() later in this method
+	fmt.Printf("Starting transcode via: %s\n", transcodeCommand.String())
+	err = transcodeCommand.Start()
+	if err != nil {
+		return fmt.Errorf("Failed to start MistProcLivepeer: %s", err)
+	}
+
+	go func() {
+		if streamOutput(stdoutPipe, &stdout, os.Stdout) != nil {
+			fmt.Errorf("Failed to stream output from stdout")
+		}
+	}()
+	go func() {
+		if streamOutput(stderrPipe, &stderr, os.Stderr) != nil {
+			fmt.Errorf("Failed to stream output from stderr")
+		}
+	}()
+
+	// TODO: remove when Mist code is updated https://github.com/DDVTECH/mistserver/issues/81
+	// Starting SOURCE_PREFIX stream because MistProcLivepeer is unable to start it automatically
+	file, err := os.CreateTemp("", "tmp-*.ts")
+	if err != nil {
+		return fmt.Errorf("error creating temporary file for SOURCE_PREFIX stream: %s", err)
+	}
+
+	if err := mistClient.PushStart(inputStream, file.Name()); err != nil {
+		return fmt.Errorf("PushStart(inputStream): %s", err)
+	}
+
+	inputUrl, err := url.Parse(request.SourceFile)
+	if err != nil {
+		return fmt.Errorf("invalid request source_location: %s", err)
+	}
+
+	dir, _ := url.Parse(".")
+	uploadDir := inputUrl.ResolveReference(dir)
+	// Cache the stream data, later used in the trigger handlers called by Mist
+	cache.DefaultStreamCache.Transcoding.Store(renditionsStream, cache.SegmentInfo{
+		CallbackUrl: request.CallbackUrl,
+		Source:      request.SourceFile,
+		Profiles:    request.Profiles[:],
+		UploadDir:   uploadDir.String(),
+	})
+
+	if err := transcodeCommand.Wait(); err != nil {
+		if exit, ok := err.(*exec.ExitError); ok {
+			log.Printf("MistProcLivepeer returned %d", exit.ExitCode())
+		}
+		return fmt.Errorf("exec transcodeCommand: %s", err)
+	}
```

Comment:
Should be started in new goroutine so HTTP API call can return with success. Studio should rely on callbacks for results on pipeline completion & errors.

Reply:
Yes, agree - I'm tempted to leave it like this for us to do some initial tests in canary environment first and then turn this into a goroutine -- mainly because it's a lot easier to debug and root cause this way.